tools/lsp/preview/properties.rs

```diff
@@ -569,151 +518,67 @@ fn create_workspace_edit_for_set_binding_on_known_property(
                     }
                 },
             };
-            common::create_workspace_edit(uri, version, vec![edit])
+            common::create_text_document_edit(uri, version, vec![edit])
         },
     )
 }
 
-fn set_binding_on_known_property(
+pub fn set_binding(
     uri: lsp_types::Url,
     version: SourceFileVersion,
     element: &common::ElementRcNode,
-    properties: &[PropertyInformation],
     property_name: &str,
-    new_expression: &str,
-    diag: &mut BuildDiagnostics,
-) -> Result<(SetBindingResponse, Option<lsp_types::WorkspaceEdit>)> {
-    let workspace_edit = if diag.has_errors() {
-        None
-    } else {
-        create_workspace_edit_for_set_binding_on_known_property(
-            uri,
-            version,
-            element,
-            properties,
-            property_name,
-            new_expression,
-        )
-    };
-
-    Ok((
-        SetBindingResponse { diagnostics: diag.iter().map(util::to_lsp_diag).collect::<Vec<_>>() },
-        workspace_edit,
-    ))
+    new_expression: String,
+) -> Option<lsp_types::WorkspaceEdit> {
+    set_binding_impl(uri, version, element, property_name, new_expression)
+        .map(|edit| common::create_workspace_edit_from_text_document_edits(vec![edit]))
 }
 
-pub fn set_binding(
-    document_cache: &common::DocumentCache,
-    uri: &lsp_types::Url,
+pub fn set_binding_impl(
+    uri: lsp_types::Url,
     version: SourceFileVersion,
     element: &common::ElementRcNode,
     property_name: &str,
     new_expression: String,
-) -> Result<(SetBindingResponse, Option<lsp_types::WorkspaceEdit>)> {
-    let (mut diag, expression_node) = {
-        let mut diagnostics = BuildDiagnostics::default();
-
-        let syntax_node = i_slint_compiler::parser::parse_expression_as_bindingexpression(
```

Comment:
If that buggy function is no longer used, it can as well be removed and from the parser